python_modules/dagster/dagster/_serdes/serdes.py

```diff
@@ -883,18 +910,63 @@ def _pack_value(
     # handle more expensive and uncommon abc instance checks last
     if isinstance(val, collections.abc.Mapping):
         return {
-            key: _pack_value(value, whitelist_map, f"{descent_path}.{key}")
+            key: _prepare_value(value, whitelist_map, object_handler, f"{descent_path}.{key}")
             for key, value in val.items()
         }
     if isinstance(val, collections.abc.Sequence):
         return [
-            _pack_value(item, whitelist_map, f"{descent_path}[{idx}]")
+            _prepare_value(item, whitelist_map, object_handler, f"{descent_path}[{idx}]")
             for idx, item in enumerate(val)
         ]
 
     raise SerializationError(f"Unhandled value type {tval}")
 
 
+def _pack_object(
+    obj: SerializableObject, whitelist_map: WhitelistMap, descent_path: str
+) -> Mapping[str, JsonSerializableValue]:
+    # the object_handler for _prepare_value to produce dicts for objects
+
+    klass_name = obj.__class__.__name__
+    serializer = whitelist_map.get_object_serializer(klass_name)
+    return dict(serializer.pack_items(obj, whitelist_map, _pack_object, descent_path))
```

Comment:
[1]

Reply:
Got it. Should we use the word "traverse" in the name of `_prepare_value`? I think that would clarify things quite a bit.